listener.go

```diff
@@ -79,6 +79,16 @@ func (l *listener) Accept() (tpt.CapableConn, error) {
 }
 
 func (l *listener) setupConn(sess quic.Session) (*conn, error) {
+	// cancel active hole punching if any
+	hpkey := sess.RemoteAddr().String()
+	l.transport.holePunchingMx.Lock()
+	cancel, ok := l.transport.holePunching[hpkey]
+	if ok {
+		cancel()
+		delete(l.transport.holePunching, hpkey)
```

Comment:
No need to `delete` here. Cleanup will be done by the `holePunch` function.

Reply:
we have too, so that the holePunch deferral doesn't double-cancel.